src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs

```diff
@@ -120,26 +120,59 @@ internal sealed class Arguments : TransformInputBase
             public TextNormKind VectorNormalizer = TextNormKind.L2;
         }
 
-        public sealed class Settings
+        /// <summary>
+        /// Advanced settings for the <see cref="TextFeaturizingEstimator"/>.
+        /// </summary>
+        public sealed class Options
```

Comment:
>Options [](start = 28, length = 7)

can you remove this, and use the Arguments above, like for everything else?

make the Factories internal, so they are not visible.

Reply:
I think we don't want to do that.
Main reason why it's bad - is bunch of Factories we don't want to expose.

---
In reply to: [254358363](https://github.com/dotnet/machinelearning/pull/2394#discussion_r254358363) [](ancestors = 254358363)

Reply:
perhaps we should do some of this cleanup outside purview of this PR ... 

I need to understand properly how the factories are used etc.

---
In reply to: [254461580](https://github.com/dotnet/machinelearning/pull/2394#discussion_r254461580) [](ancestors = 254461580,254358363)